### PR TITLE
add some waypoint related E2E test cases

### PR DESCRIPTION
**Pull request by YaoZengzeng**
**What type of PR is this?**

/kind enhancement

**What this PR does / why we need it**:

migrate some waypoint related E2E test cases

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: test/e2e/baseline_test.go [hunk 1]
```diff
@@ -138,12 +140,228 @@ spec:
...
+			}
+
+			shouldBalance := dst.Config().HasServiceAddressedWaypointProxy()
+			// Istio client will not reuse connections for HTTP/1.1
+			opt.HTTP.HTTP2 = true
+			// Make sure we make multiple calls
+			opt.Count = 10
+			c := singleHost
+			if shouldBalance {
+				c = multipleHost
```
Comment (hzxuzhonghu) [inline test/e2e/baseline_test.go:185]: do we have workload level waypoint?

File: test/e2e/baseline_test.go [hunk 2]
```diff
@@ -138,12 +140,228 @@ spec:
...
+metadata:
+  name: route
+spec:
+  hosts:
+  - "{{.Destination}}"
+  http:
+  - headers:
+      request:
+        add:
+          istio-custom-header: user-defined-value
```
Comment (hzxuzhonghu) [inline test/e2e/baseline_test.go:215]: should update to kmesh

File: test/e2e/baseline_test.go [hunk 3]
```diff
@@ -138,12 +140,228 @@ spec:
...
+    route:
+    - destination:
+        host: "{{.Destination}}"
+`).ApplyOrFail(t)
+				opt.Check = check.And(
+					check.OK(),
+					check.RequestHeader("Istio-Custom-Header", "user-defined-value"))
+				src.CallOrFail(t, opt)
+			})
+			t.NewSubTest("subset").Run(func(t framework.TestContext) {
```
Comment (hzxuzhonghu) [inline test/e2e/baseline_test.go:225]: replace `subset` with a more meaningful name

**Review by hzxuzhonghu — COMMENTED** [review]
Since most codes are from istio, we need to add some document, suggest add a readme in e2e directory

File: test/e2e/baseline_test.go (hunk 1 above)
Comment (YaoZengzeng) [reply test/e2e/baseline_test.go:185]: Not yet, relevant test cases will be added later

**Comment by YaoZengzeng** [comment]
@hzxuzhonghu updated

**Review by hzxuzhonghu — APPROVED** [review]
/lgtm

**Pull request merged**